src/modules/job-manager/event.c

```diff
@@ -601,18 +614,25 @@ int event_job_post_pack (struct event *event,
     int saved_errno;
     double timestamp;
     flux_job_state_t old_state = job->state;
+    int eventlog_seq = EVENT_JOURNAL_ONLY & flags ? -1 : job->eventlog_seq;
```

Comment:
I think you missed flipping flags and constant, and probably should also put parens around the logical expressin.